### PR TITLE
merge Feature/3495 to master(New annotator framework)

### PR DESCRIPTION
**Pull request by bartcharbon**

### CODE REVIEW

File: molgenis-data-mysql/src/test/java/org/molgenis/data/mysql/MysqlRepositoryCountTest.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public void test()
 		Assert.assertEquals(persons.count(new QueryImpl().eq("birthday", "1950-01-31")), 1);
 		Assert.assertEquals(persons.count(new QueryImpl().gt("birthday", "1950-01-31")), 2);
 		Assert.assertEquals(
-				persons.count(new QueryImpl().gt("birthday", "1976-06-07").or().lt("birthday", "1976-06-07")), 2);
```
Comment (bartcharbon) [inline molgenis-data-mysql/src/test/java/org/molgenis/data/mysql/MysqlRepositoryCountTest.java]: accidental commit, will fix

**Comment by Mark-de-Haan** [comment]
If I change the location of a resource via the Runtime properties, I have to restart the app before the new location is detected

**Comment by Mark-de-Haan** [comment]
Please merge with master, the molgenis-catalogue module has been removed.

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/cmd/CmdLineAnnotator.java [hunk 2]
```diff
@@ -15,134 +20,153 @@
...
-			for (String ann : annotators)
-			{
-				System.out.print(ann + " ");
-			}
-			throw new Exception("\nInvalid annotator.\n" + "Possible annotators are: " + annotators.toString() + ".");
+			System.out.println("Annotator must be one of the following: " + annotatorNames.toString());
+			return;
 		}
 
+		// EntityAnnotator annotator = annotationService.getAnnotatorByName(annotatorName);
```
Comment (Mark-de-Haan) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/cmd/CmdLineAnnotator.java]: remove commented code

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/AnnotationStep.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.Entity;
+
+/**
+ * A step to use in the Annotator. Each step has requirements on the annotated entity's metadata.
+ * 
+ * @author fkelpin
+ *
+ */
+public interface AnnotationStep
```
Comment (Mark-de-Haan) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/AnnotationStep.java]: Requires new name to better specify its function: Classes that implement can be asked; Which attributes do you need?

Comment (fdlk) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/AnnotationStep.java]: AttributeReader?

**Pull request merged**